pkg/net/coap/auth.go

```diff
@@ -0,0 +1,48 @@
+package coap
```

Comment:
I am in favor of having a repository structure that structures things based on their primary functionality of domain in the business logic. That means, and it's for 95% possible, place these "utils" to the package where they from business perspective belong to.
In this case, coap-gateway.

Reply:
moved to coap-gw